tests/test_functional.py

```diff
@@ -0,0 +1,130 @@
+        web_profile_key = next(profile["key"] for profile in web_profiles["profiles"])
+        web_rule = self._search_rules(activation="true", qprofile=web_profile_key, rule_key=overridden_key)
+        self.assertEqual(1, web_rule["total"])
+
+        # Verify that it indeed differs from the Sonar way
+        sonar_way_web = self._search_profiles(language="web", qualityProfile="Sonar way")
+        sonar_way_key = next(p["key"] for p in sonar_way_web["profiles"])
+        sonar_way_rule = self._search_rules(activation="false", qprofile=sonar_way_key, rule_key=overridden_key)
+        self.assertEqual(1, sonar_way_rule["total"])
+
+    def test_rule_params_in_profile(self):
+        """Check that custom profile rule params are applied."""
+        cs_param_rule_lines = [rule_line for rule_line in self.config_json["rules"]["cs"] if "|" in rule_line]
+        self.assertNotEqual([], cs_param_rule_lines)
+
+        version_profile = f"ictu-{self.config_json['profiles']['cs']['version']}-{self.config_json['rules_version']}"
+        cs_profile_search = self._search_profiles(language="cs", qualityProfile=version_profile)
+        self.assertEqual(1, len(cs_profile_search["profiles"]))
+
+        # Fetch changelog api with requests, because library does not expose it
+        changelog_api = f"{self.sonar_api}/qualityprofiles/changelog"
+        change_history_params = {"language": "cs", "qualityProfile": version_profile, "ps": 500}
+        api_result = requests.get(changelog_api, auth=self.sonar_auth, params=change_history_params).json()
+        self.assertGreater(change_history_params["ps"], api_result["total"])
+
+        # Verify that each param override has a corresponding changelog event
+        profile_changes = api_result["events"]
+        for cs_rule_line in cs_param_rule_lines:
+            rule_key, rule_params_str = cs_rule_line[1:].split()[0].split("|")
+            rule_changes = [change for change in profile_changes if change["ruleKey"] == rule_key]
+            for rule_param_str in rule_params_str.split(";"):
+                rule_params = rule_param_str.split("=")
+                param_dict = {rule_params[0]: rule_params[1]}
+                self.assertTrue(next(
+                    param_dict.items() <= rule_change.get('params', {}).items() for rule_change in rule_changes
+                ))
+
+        # Verify that it indeed differs from the Sonar way
+        self.assertTrue(next(event for event in profile_changes if event["action"] == "UPDATED"))
+
+    def test_profile_rule_deactivation(self):
+        """Check that custom profile rule deactivation is applied."""
+        ts_rule_lines = [rule_line for rule_line in self.config_json["rules"]["ts"] if "types=" not in rule_line]
+        self.assertTrue(any([rule_line.startswith("-") for rule_line in ts_rule_lines]))
+        self.assertTrue(any([rule_line.startswith("+") for rule_line in ts_rule_lines]))
```

Comment:
Does this check for activation? If so, change name of function (only deactivation)?

Reply:
This verifies the assumption that there are both activation and deactivation rules in the config.
From a naming point of view it might be handy to split the tasks, but it would create a lot of duplication